mcs/class/System/Test/System.Net/HttpWebRequestTest.cs

```diff
@@ -303,7 +303,8 @@ public void BeginGetRequestStream_Body_NotAllowed ()
 				request.Method = "GET";
 
 				try {
-					request.BeginGetRequestStream (null, null);
+					var result = request.BeginGetRequestStream (null, null);
+					request.EndGetRequestStream (result);
```

Comment:
The test was broken, there is on guarantee that `Begin*()` will actually do something synchronously, you should always call the `End*()` method as well to get the exception.
In the new code, `Begin*()` and `End*()` are just wrappers around `*Async()`.